test/e2e/framework.go

```diff
@@ -3220,3 +3221,74 @@ func (data *TestData) GetPodLogs(ctx context.Context, namespace, name, container
 	}
 	return b.String(), nil
 }
+
+func (data *TestData) runDNSQuery(
+	podName string,
+	containerName string,
+	podNamespace string,
+	dstAddr string,
+	useTCP bool,
+	dnsServiceIP string) (net.IP, error) {
+
+	digCmd := fmt.Sprintf("dig "+"@"+dnsServiceIP+" +short %s", dstAddr)
+	if useTCP {
+		digCmd += " +tcp"
+	}
+	cmd := []string{
+		"/bin/sh",
+		"-c",
+		digCmd,
+	}
```

Comment:
Are you sure you need to wrap this command with a shell invocation?
I think you don't, it is usually better to call the executable directly (in this case, `dig`), unless you need to use shell-specific utilities (shell built-ins, piping, ...).